File: java-sdks/src/main/java/com/suisdks/sui/cryptography/BouncyCastleProviderHolder.java
package com.suisdks.sui.cryptography;

import java.security.Security;
import org.bouncycastle.jce.provider.BouncyCastleProvider;

public final class BouncyCastleProviderHolder {
    private BouncyCastleProviderHolder() {
    }

    public static void ensureInstalled() {
        if (Security.getProvider("BC") == null) {
            Security.addProvider(new BouncyCastleProvider());
        }
    }
}

File: kotlin-sdks/src/main/kotlin/com/suisdks/sui/cryptography/BouncyCastleProviderHolder.kt
package com.suisdks.sui.cryptography

import java.security.Security
import org.bouncycastle.jce.provider.BouncyCastleProvider

object BouncyCastleProviderHolder {
    fun ensureInstalled() {
        if (Security.getProvider("BC") == null) {
            Security.addProvider(BouncyCastleProvider())
        }
    }
}
